app/models/form/sales/pages/organisation.rb

```diff
@@ -10,24 +10,40 @@ def questions
     ]
   end
 
-  def routed_to?(_log, current_user)
-    if FeatureToggle.merge_organisations_enabled?
+  def routed_to?(log, current_user)
+    return false unless current_user
+    return true if current_user.support?
+    return false unless FeatureToggle.sales_managing_organisation_enabled?
 
-      return false unless current_user
-      return true if current_user.support?
+    return true if stock_owners_with_own_stock_count(current_user) > 1
+    return true if current_user.organisation.holds_own_stock? && stock_owners_with_own_stock_count(current_user) >= 1
 
-      absorbed_stock_owners = current_user.organisation.absorbed_organisations.where(holds_own_stock: true)
+    stock_owners = if FeatureToggle.merge_organisations_enabled?
+                     current_user.organisation.stock_owners.where(holds_own_stock: true) + current_user.organisation.absorbed_organisations.where(holds_own_stock: true)
+                   else
+                     current_user.organisation.stock_owners.where(holds_own_stock: true)
+                   end
 
-      if current_user.organisation.holds_own_stock?
-        return true if absorbed_stock_owners.count >= 1
-      else
-        return false if absorbed_stock_owners.count.zero?
-        return true if absorbed_stock_owners.count > 1
+    if current_user.organisation.holds_own_stock?
+      if FeatureToggle.merge_organisations_enabled? && current_user.organisation.absorbed_organisations.any?(&:holds_own_stock?)
+        return true
       end
+      return true if stock_owners.count >= 1
 
-      false
+      log.update!(owning_organisation: current_user.organisation)
     else
-      !!current_user&.support?
+      return false if stock_owners.count.zero?
+      return true if stock_owners.count > 1
+
+      log.update!(owning_organisation: stock_owners.first)
     end
+
+    false
+  end
+
+private
+
+  def stock_owners_with_own_stock_count(user)
```

Comment:
Can we move this out of the page to a shared location for sales and lettings? Think I might have been the one to put the equivalent method on the lettings page, sorry if that's true!

Reply:
Ended up removing this altogether